<?xml version="1.0"?>

<!-- map.launch -->
<launch>
    <!-- Competition Waypoints -->
    <arg name="simulation" default="false"/>
    <include file="$(find igvc_navigation)/launch/waypoint_source.launch"
             unless="$(arg simulation)"/>


     <!-- Master -->
     <node  name = "pather" pkg="igvc_navigation" type="pather" output="screen">
         <!-- action timeout duration (seconds)-->
         <param name="max_action_time" type="double" value="20.0"/>
     </node>

    <!-- Trajectory planning service -->
    <node name="generate_path_server" pkg="igvc_navigation" type="generate_path_server" output="screen" >
        <param name="goal_threshold" type="double" value="0.8"/>
        <param name="c_space" type="double" value="0.5"/>
        <param name="probability_threshold" type="double" value="0.7"/>
        <param name="point_turns_enabled" type="bool" value="false"/>
        <param name="reverse_enabled" type="bool" value="false"/>
        <param name="max_jump_size" type="double" value="10.0"/>
        <param name="theta_filter" type="double" value="1.5"/>
        <param name="max_theta_change" type="double" value="0.8"/>
        <param name="theta_change_window" type="double" value="3.0"/>
    </node>

    <!-- Path-follow action server -->
    <node name="path_follow_server" pkg="igvc_navigation" type="path_follow_server" output="screen" >
        <param name="maximum_vel" type="double" value="6.0"/>
        <param name="target_v" type="double" value="0.6"/>
        <param name="axle_length" type="double" value="0.52"/>
        <param name="lookahead_dist" type="double" value="1.8"/>
        <param name="k1" type="double" value="0.5"/>
        <param name="k2" type="double" value="1.0"/>
    </node>



</launch>
